velox/functions/prestosql/types/IPPrefixType.h

```diff
@@ -97,55 +48,41 @@ tryParseIpPrefixString(folly::StringPiece ipprefixString) {
         threadSkipErrorDetails()
             ? Status::UserError()
             : Status::UserError(
-                  "Invalid CIDR IP address specified. Expected IP/PREFIX format, got: {}",
-                  ipprefixString));
+                  "Cannot cast value to IPPREFIX: {}", ipprefixString));
   }
 
   auto tryCdirNetwork = folly::IPAddress::tryCreateNetwork(
       ipprefixString, /*defaultCidr*/ -1, /*applyMask*/ false);
+
   if (tryCdirNetwork.hasError()) {
-    return folly::makeUnexpected(handleFailedToCreateNetworkError(
-        ipprefixString, std::move(tryCdirNetwork.error())));
+    return folly::makeUnexpected(
+        threadSkipErrorDetails()
+            ? Status::UserError()
+            : Status::UserError(
+                  "Cannot cast value to IPPREFIX: {}", ipprefixString));
   }
 
-  folly::ByteArray16 addrBytes;
-  const auto& cdirNetwork = tryCdirNetwork.value();
-  if (cdirNetwork.first.isIPv4Mapped() || cdirNetwork.first.isV4()) {
-    // Validate that the prefix value is <= 32 for ipv4
-    if (cdirNetwork.second > ipaddress::kIPV4Bits) {
-      return folly::makeUnexpected(
-          threadSkipErrorDetails()
-              ? Status::UserError()
-              : Status::UserError(
-                    "CIDR value '{}' is > network bit count '{}'",
-                    cdirNetwork.second,
-                    ipaddress::kIPV4Bits));
-    }
-    auto ipv4Addr = folly::IPAddress::createIPv4(cdirNetwork.first);
-    auto ipv4AddrWithMask = ipv4Addr.mask(cdirNetwork.second);
-    auto ipv6Addr = ipv4AddrWithMask.createIPv6();
-    addrBytes = ipv6Addr.toByteArray();
-  } else {
-    // Validate that the prefix value is <= 128 for ipv6
-    if (cdirNetwork.second > ipaddress::kIPV6Bits) {
-      return folly::makeUnexpected(
-          threadSkipErrorDetails()
-              ? Status::UserError()
-              : Status::UserError(
-                    "CIDR value '{}' is > network bit count '{}'",
-                    cdirNetwork.second,
-                    ipaddress::kIPV6Bits));
-    }
-    auto ipv6Addr = folly::IPAddress::createIPv6(cdirNetwork.first);
-    auto ipv6AddrWithMask = ipv6Addr.mask(cdirNetwork.second);
-    addrBytes = ipv6AddrWithMask.toByteArray();
+  auto [ip, prefix] = tryCdirNetwork.value();
+  if (prefix > ((ip.isIPv4Mapped() || ip.isV4()) ? ipaddress::kIPV4Bits
+                                                 : ipaddress::kIPV6Bits)) {
+    return folly::makeUnexpected(
+        threadSkipErrorDetails()
+            ? Status::UserError()
+            : Status::UserError(
+                  "Cannot cast value to IPPREFIX: {}", ipprefixString));
   }
 
+  folly::ByteArray16 addrBytes;
   int128_t intAddr;
-  // Similar to IPAdressType, assume Velox is always on little endian systems
+  addrBytes = (ip.isIPv4Mapped() || ip.isV4())
```

Comment:
You can combine line 75 and this one together and declare and initialize addrBytes.
And you can use `auto` for the type too.

Also you'd ant to move the declaration of `intAddr` before line 82. Best to initialize it to 0 too.

Reply:
Done!